### PR TITLE
BaSD-MS-51: Implement React Final Form In Clients

### PR DESCRIPTION
**Pull request by luisnannini**
Task:
- Modify form to use react final form in clients component
- Add required function
- Add validate function
- Use submitting in Fields and buttons
- Use pristine in buttons

### CODE REVIEW

**Review by GuidoGlielmi — COMMENTED** [review]
Update development branch and merge it into your branch so vercel accepts it

File: src/Components/Admin/Clients/Form/index.js [hunk 1]
```diff
@@ -43,181 +44,219 @@ function Form() {
...
+  // const onChangeLogoInput = (event) => {
+  //   setLogoValue(event.target.value);
+  // };
 
-  const onChangeDescriptionInput = (event) => {
-    setDescriptionValue(event.target.value);
-  };
+  // const onChangeDescriptionInput = (event) => {
+  //   setDescriptionValue(event.target.value);
+  // };
```
Comment (nicolasblasco) [inline src/Components/Admin/Clients/Form/index.js:77]: @luisnannini we should erase commented code

File: src/Components/Admin/Clients/Form/index.js [hunk 2]
```diff
@@ -43,181 +44,187 @@ function Form() {
...
+    // Phone Number
+    if (!formValues.phone?.toString().match(/^\d+$/)) {
+      errors.phone = 'Phone number must contain only numbers';
+    }
+    if (formValues.phone?.length < 7 || formValues.phone?.length > 14) {
+      errors.phone = 'Phone number must be between 7 and 14 numbers';
+    }
+    // Country
+    if (formValues.country?.length < 5) {
+      errors.country = 'School must contain at least 5 characters';
```
Comment (nicolasblasco) [inline src/Components/Admin/Clients/Form/index.js:107]: @luisnannini we should change school to be country and maybe be at least 3 characters

File: src/Components/Admin/Clients/Form/index.js [hunk 3]
```diff
@@ -43,181 +44,187 @@ function Form() {
...
+    }
+    if (formValues.country?.length > 50) {
+      errors.country = 'School must be less than 50 characters';
```
Comment (nicolasblasco) [inline src/Components/Admin/Clients/Form/index.js:110]: @luisnannini we should change school to be country

File: src/Components/Admin/Clients/Form/index.js [hunk 4]
```diff
@@ -43,181 +44,187 @@ function Form() {
...
+    // State
+    if (formValues.state?.length < 3) {
+      errors.state = 'State must contain at least 3 characters';
+    }
+    if (formValues.state?.length > 50) {
+      errors.state = 'State must be less than 50 characters';
+    }
+    return errors;
+  };
+
```
Comment (nicolasblasco) [inline src/Components/Admin/Clients/Form/index.js:128]: @luisnannini we should also add a validation for the description

File: src/Components/Admin/Clients/Form/index.js [hunk 5]
```diff
@@ -43,181 +44,187 @@ function Form() {
...
+                  initialValue={descriptionValue}
+                  placeholder={'Insert Client description...'}
+                  disabled={isLoading}
+                  component={Input}
+                />
+              </div>
+            </div>
+            <div className={styles.button}>
+              <ButtonCancel disabled={isLoading} onClick={() => history.push('/admin/clients')} />
+              <ButtonConfirm disabled={isLoading} type="submit" />
```
Comment (nicolasblasco) [inline src/Components/Admin/Clients/Form/index.js:220]: @luisnannini we should change disabled to have formProps submitting a pristine

**Review by nicolasblasco — CHANGES_REQUESTED** [review]
@luisnannini implementation of final form is very good, but we should see the comments requested before approving

File: src/Components/Admin/Clients/Form/index.js [hunk 6]
```diff
@@ -43,181 +44,187 @@ function Form() {
...
+                  component={Input}
+                  validate={required}
+                />
+                <Field
+                  label={'Address'}
+                  name={'address'}
+                  type={'address'}
+                  initialValue={addressValue}
+                  placeholder={'Insert Client address...'}
+                  disabled={isLoading}
```
Comment (nicolasblasco) [inline src/Components/Admin/Clients/Form/index.js:157]: @luisnannini we should change disabled to have formProps.submitting (in all fields)

**Review by nicolasblasco — APPROVED** [review]

**Review by ValentinaCollareta — APPROVED** [review]

**Pull request merged**